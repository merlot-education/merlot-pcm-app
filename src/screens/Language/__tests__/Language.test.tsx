import React from 'react';
import { create } from 'react-test-renderer';
import Language from '../Language';

jest.mock('react-native-localize', () => ({
  findBestAvailableLanguage: () => ({
    languageTag: 'en-US',
    isRTL: false,
  }),
}));

jest.mock('i18next', () => ({
  use: () => {
    return {
      init: jest.fn(),
    };
  },
  init: jest.fn(),
  t: k => k,
  language: 'en',
}));

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        language: 'en',
        changeLanguage: () => new Promise(jest.fn()),
      },
    };
  },
}));

describe('Language', () => {
  it('should render correctly', () => {
    const tree = create(<Language />);

    expect(tree).toMatchSnapshot();
  });
});
