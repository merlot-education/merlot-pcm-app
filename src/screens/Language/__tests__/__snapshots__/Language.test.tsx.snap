// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language should render correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
    }
  }
  testID="safeareaview"
>
  <View>
    <View
      style={
        Object {
          "padding": 20,
          "width": "100%",
        }
      }
      testID="SingleSelectBlock"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#CCE8F4",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          accessibilityLabel="Text"
          style={
            Array [
              Object {
                "color": "#000094",
              },
              undefined,
            ]
          }
          testID="Text"
        >
          Language.English
        </Text>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000094",
                "fontSize": 25,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î—Š
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#CCE8F4",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          accessibilityLabel="Text"
          style={
            Array [
              Object {
                "color": "#000094",
              },
              undefined,
            ]
          }
          testID="Text"
        >
          Language.French
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#CCE8F4",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          accessibilityLabel="Text"
          style={
            Array [
              Object {
                "color": "#000094",
              },
              undefined,
            ]
          }
          testID="Text"
        >
          Language.German
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
